wodles/aws/tests/test_server_access.py

```diff
@@ -0,0 +1,229 @@
+                        mock_debug.assert_any_call(
+                            f"++ File previously processed, but reparse flag set: {bucket_file['Key']}",
+                            1)
+                    else:
+                        mock_debug.assert_any_call(f"++ Skipping previously processed file: {bucket_file['Key']}", 2)
+                        continue
+
+                    mock_debug.assert_any_call(f"++ Found new log: {bucket_file['Key']}", 2)
+                    mock_get_log_file.assert_called_with(aws_account_id, bucket_file['Key'])
+                    mock_iter_events.assert_called()
+
+                    if instance.delete_file:
+                        mock_debug.assert_any_call(f"+++ Remove file from S3 Bucket:{bucket_file['Key']}", 2)
+
+                    mock_mark_complete.assert_called_with(aws_account_id, aws_region, bucket_file)
+
+                if object_list['IsTruncated']:
+                    mock_build_filter.assert_any_call(aws_account_id, aws_region, True)
+
+
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+def test_aws_server_access_iter_files_in_bucket_handles_exceptions(mock_sts):
+    """Test 'iter_files_in_bucket' method handles exceptions raised when the filename does not have the valid format
+    or by an unexpected cause and exits with the expected exit code.
+    """
+    instance = utils.get_mocked_bucket(class_=server_access.AWSServerAccess)
+
+    instance.client = MagicMock()
+
+    with patch('aws_bucket.AWSBucket.build_s3_filter_args') as mock_build_filter:
+        with pytest.raises(SystemExit) as e:
+            instance.skip_on_error = False
+            utils.LIST_OBJECT_V2_NO_PREFIXES['Contents'][0]['Key'] = ['123']
+            instance.client.list_objects_v2.return_value = utils.LIST_OBJECT_V2_NO_PREFIXES
+            instance.iter_files_in_bucket(utils.TEST_ACCOUNT_ID, utils.TEST_REGION)
+        assert e.value.code == utils.INVALID_KEY_FORMAT_ERROR_CODE
+
+        with pytest.raises(SystemExit) as e:
+            mock_build_filter.side_effect = Exception
+            instance.iter_files_in_bucket(utils.TEST_ACCOUNT_ID, utils.TEST_REGION)
+        assert e.value.code == utils.UNEXPECTED_ERROR_WORKING_WITH_S3
+
+
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+def test_aws_server_access_marker_only_logs_after(mock_sts):
+    """Test 'marker_only_logs_after' method returns the expected marker using the `only_logs_after` value."""
+    test_only_logs_after = utils.TEST_ONLY_LOGS_AFTER
+
+    instance = utils.get_mocked_bucket(class_=server_access.AWSServerAccess, only_logs_after=test_only_logs_after)
+    instance.prefix = utils.TEST_PREFIX
+
+    instance.date_format = '%Y-%m-%d'
+
+    marker = instance.marker_only_logs_after(aws_region=utils.TEST_REGION, aws_account_id=utils.TEST_ACCOUNT_ID)
+    assert marker == f"{instance.prefix}{test_only_logs_after[0:4]}-{test_only_logs_after[4:6]}-{test_only_logs_after[6:8]}"
```

Comment:
Some lines exceed 120 characters, which is the value most of our API and framework files take:
![image](https://github.com/wazuh/wazuh/assets/23361101/b2388387-511c-49df-bdc1-2bf5c48f4b06)